### PR TITLE
Oea sadness

### PR DESCRIPTION
**Pull request by bracken**

### CODE REVIEW

File: client/js/components/main/attempts.jsx [hunk 1]
```diff
@@ -39,15 +39,18 @@ export default class Attempts extends BaseComponent{
   attemptsStuff(ua){
     var that = this;
     return <div>
-      {ua.attempts.map(function(attempt){
-        var score = 'none',
+      {ua.attempts.slice().reverse().map(function(attempt){
```
Comment (levionessa) [inline client/js/components/main/attempts.jsx]: What is `ua` and will `ua.attempts` ever be `nil`?

File: client/js/stores/assessment.js [hunk 2]
```diff
@@ -69,7 +69,8 @@ function selectAnswer(answer){
   else if (item.question_type == 'essay_question') {
     _selectedAnswerIds = answer;
   }
-  else if (item.question_type = 'multiple_dropdowns_question') {
+  else if (item.question_type == 'multiple_dropdowns_question') {
```
Comment (levionessa) [inline client/js/stores/assessment.js]: Good catch! :clap:

File: client/js/components/main/attempts.jsx (hunk 1 above)
Comment (bracken) [reply client/js/components/main/attempts.jsx]: If there is a record of the learner on this page it should never be null. That would have been a problem for `#map` as well and I don't think we've had a problem. But good question, I'm only mostly confident. 

You probably figured it out, but that `#slice` is just so that the array itself isn't modified with the `#reverse` call. I googled it and it's what a few javascripters suggested even though it makes me feel funny.

**Comment by levionessa** [comment]
It's hard to grasp some of the changes around checking current state especially without a much deeper dive and context. A few general things not so much about the actual diff:

* I haven't used CodeClimate in forever. Is there a way to globally ignore the error about duplicate code in two places? This is one thing they've always gotten way wrong that's bothered me. Duplicate code != not DRY. DRY is about centralizing changes and sometimes duplicating code is *exactly* what you want to do to do that (i.e. "these two things have the same implementation now but abstracting them into one thing conflates concerns" or "these two things have the same implementation now and that sorta works in a hacky way but we *know* they'll be different and need to change independently."

* I really appreciate the test plans in commit bodies. I'd *love* in this quarter to try to prioritize at least some first steps towards UI testing. I'm concerned about how much effort it takes us as a team to manually QA things, how much we're likely to miss by hand, and especially code clarity / understanding when to refactor. Especially since there's not a lot of Javascript *expertise* here, some tests would really help us notice as a team places where code is hard to understand / test / maintain.

* Other than code climate, we aren't using any linter or code quality tools as part of our regular test suite, correct? I'd like to change that if that is correct.

**Comment by levionessa** [comment]
TL;DR it's hard for me to with 100% confidence claim this is or isn't a risky change but it all looks fine and minimal tome and your verbose commit messages really helped me understand the changes and the test strategy. :clap: Totally good with this as is.
